### PR TITLE
Move user update requests from the client to back-end

### PR DESCRIPTION
**Pull request by sergous**
https://github.com/dtpstat/website/issues/44

* Create custom api call me for auth0
* Move user update request from user-profile-provider
* Update all useUser usages
* Modify auth0 me output for include user format

Снимок экрана 2022-01-31 в 5 30 12 AM

### CODE REVIEW

File: src/pages/iframes/comments.tsx [hunk 1]
```diff
@@ -14,13 +14,12 @@ import {
   IframeResizerScript,
 } from "../../shared/django-helpers";
 
+// Keeping "height: 100%" from src/styles/inherited-scss/helpers/_base.scss
+// would invalidate iframe resizing
+// @todo: update global styles and remove
```
Comment (kachkaev) [inline src/pages/iframes/comments.tsx:19]: Out of curiosity: what made you move these comments? AFAIU, `//` comments are removed from `styled-components` styles on build.

**Review by kachkaev — APPROVED** [review]
I like the new approach better!

We can see if we can avoid `auth0UserProfile.user as User`, but we can try something in a follow-up PR.

File: src/pages/iframes/comments.tsx (hunk 1 above)
Comment (sergous) [reply src/pages/iframes/comments.tsx:19]: As you can see on the screenshot this comments breaks the coloring in vscode.

Снимок экрана 2022-02-01 в 6 49 22 AM

Comment (kachkaev) [reply src/pages/iframes/comments.tsx:19]: Interesting! Looks like a bug in VScode. You can try installing or updating the styled-components extension. But happy for us to keep comments outside in this case.

Comment (sergous) [reply src/pages/iframes/comments.tsx]: I've figured out - only the quotes in the comment are making the mess. I've removed it, works fine now.

**Review by kachkaev — APPROVED** [review]

**Pull request merged**